Makefile

```diff
@@ -1,29 +1,159 @@
-TARGET_BPF := test/test.bpf.o
-VMLINUX_H = test/vmlinux.h
+BASEDIR = $(abspath ./)
 
-GO_SRC := $(shell find . -type f -name '*.go')
-BPF_SRC := $(shell find . -type f -name '*.bpf.c')
-PWD := $(shell pwd)
+OUTPUT = ./output
+SELFTEST = ./selftest
 
-LIBBPF_HEADERS := /usr/include/bpf
-LIBBPF := "-lbpf"
+CC = gcc
+CLANG = clang
 
-.PHONY: all
-all: test
+BTFFILE = /sys/kernel/btf/vmlinux
+BPFTOOL = $(shell which bpftool || /bin/false)
+VMLINUX_H = $(abspath $(OUTPUT)/vmlinux.h)
 
-$(VMLINUX_H):
-	bpftool btf dump file /sys/kernel/btf/vmlinux format c > test/vmlinux.h
+# libbpf
 
-go_env := CC=gcc CGO_CFLAGS="-I $(LIBBPF_HEADERS)" CGO_LDFLAGS="$(LIBBPF)"
-.PHONY: test
-test: $(TARGET_BPF) $(GO_SRC)
-	$(go_env) go test -ldflags '-extldflags "-static"' .
+LIBBPF_SRC = $(abspath ./libbpf/src)
+LIBBPF_OBJ = $(abspath ./$(OUTPUT)/libbpf.a)
+LIBBPF_OBJDIR = $(abspath ./$(OUTPUT)/libbpf)
+LIBBPF_DESTDIR = $(abspath ./$(OUTPUT))
 
-$(TARGET_BPF): $(BPF_SRC) $(VMLINUX_H)
-	clang \
-		-g -O2 -c -target bpf \
-		-o $@ $<
+CFLAGS = -g -O2 -Wall -fpie
+LDFLAGS =
 
-.PHONY: clean
-clean: 
-	rm $(TARGET_BPF) $(VMLINUX_H)
+# golang
+
+CGO_CFLAGS_STATIC = "-I$(abspath $(OUTPUT))"
+CGO_LDFLAGS_STATIC = "-lelf -lz $(LIBBPF_OBJ)"
+
+CGO_CFGLAGS_DYN = "-I. -I/usr/include/"
+CGO_LDFLAGS_DYN = "-lelf -lz -lbpf"
+
+# default == shared lib from OS package
+
+all: libbpfgo-dynamic
+install: libbpfgo-dynamic-install
+test: libbpfgo-dynamic-test
+
+# libbpfgo test object
+
+libbpfgo-test-bpf:
+	$(MAKE) -C $(SELFTEST)/build
+
+libbpfgo-test-bpf-clean:
+	$(MAKE) -C $(SELFTEST)/build clean
+
+# libbpf: shared
+
+libbpfgo-dynamic: outputlibdir
+	CC=$(CLANG) \
+	   CGO_CFLAGS=$(CGO_CFLAGS_DYN) \
+	   CGO_LDFLAGS=$(CGO_LDFLAGS_DYN) \
+	   go build .
+
+libbpfgo-dynamic-install: libbpfgo-dynamic
```

Comment:
can you please explain the purpose of this target?

Reply:
Both installs have no meaning as we currently don't have any binary being installed. I'll probably remove both and, maybe, add 'go list -json' for both as being discussed at:

https://github.com/aquasecurity/libbpfgo/pull/28#pullrequestreview-701651177